src/components/TranslateBoxes.js

```diff
@@ -3,27 +3,62 @@ import Grid from '@material-ui/core/Grid';
 import Paper from '@material-ui/core/Paper';
 import TextField from '@material-ui/core/TextField';
 import Box from '@material-ui/core/Box';
+import { Typography, Tooltip, Link } from '@material-ui/core';
 
 
 export default function TranslateBoxes({input, handleInputChange, output, errors, classes}){
-    function addErrorsToOutput(output, errors){
-        if (errors && Object.entries(errors).length !== 0){
-            let finalOutput = output
 
-            // go through output text and find any strings with pattern $$E_$$
-            const regexp = /\$\$E.\$\$/g
-            const matches = [...output.matchAll(regexp)];
+    /**
+     * Function takes output and errors and returns a list of typography and tooltip
+     * elements to be displayed in the output area. Errors are found in the output and replaced
+     * with "<err>" as well as a tooltip explaining the error.
+     * @param {response translation from backend} output 
+     * @param {response errors from backend} errors 
+     */
+    function createOutputTypography(output, errors){
+        const regexp = /\$\$E.\$\$/g
+        const matches = [...output.matchAll(regexp)];
 
+        if (matches.length > 0 && errors && Object.entries(errors).length !== 0){
+            console.log(matches)
+            let prev = 0;
+            let finalOutput = []
             for (const match of matches){
-                const errorKey = match[0].substring(2).slice(0,-2)
-                const errorMessage = errors[errorKey]["errorMessage"]
-                finalOutput = finalOutput.replace(match, errorMessage)
+                // Add text before error
+                let beforeText = output.substring(prev, match["index"])
```

Comment:
Just to double check, has this been tested on output that starts with an error?

Reply:
Yes it does!